libp2p/kad_dht/kad_dht.py

```diff
@@ -0,0 +1,545 @@
+                                for addr in addrs:
+                                    peer_proto.addrs.append(addr.to_bytes())
+                            except Exception:
+                                pass
+
+                    # Serialize and send response
+                    response_bytes = response.SerializeToString()
+                    await stream.write(varint.encode(len(response_bytes)))
+                    await stream.write(response_bytes)
+                    logger.debug("Sent GET_PROVIDERS response")
+
+                # Handle GET_VALUE message
+                elif message.type == Message.MessageType.GET_VALUE:
+                    # Process GET_VALUE
+                    key = message.key
+                    logger.debug(f"Received GET_VALUE request for key {key.hex()}")
+
+                    value = self.value_store.get(key)
+                    if value:
+                        logger.debug(f"Found value for key {key.hex()}")
+
+                        # Create response using protobuf
+                        response = Message()
+                        response.type = Message.MessageType.GET_VALUE
+
+                        # Create record
+                        response.key = key
+                        response.record.key = key
+                        response.record.value = value
+                        response.record.timeReceived = str(time.time())
+
+                        # Serialize and send response
+                        response_bytes = response.SerializeToString()
+                        await stream.write(varint.encode(len(response_bytes)))
+                        await stream.write(response_bytes)
+                        logger.debug("Sent GET_VALUE response")
+                    else:
+                        logger.debug(f"No value found for key {key.hex()}")
+
+                # Handle PUT_VALUE message
+                elif message.type == Message.MessageType.PUT_VALUE and message.HasField(
+                    "record"
+                ):
+                    # Process PUT_VALUE
+                    key = message.record.key
+                    value = message.record.value
+
+                    if key and value:
+                        self.value_store.put(key, value)
+                        logger.debug(f"Stored value for key {key.hex()}")
+
+                        # Send acknowledgement
+                        response = Message()
+                        response.type = Message.MessageType.PUT_VALUE
+                        response.key = key
+
+                        response_bytes = response.SerializeToString()
+                        await stream.write(varint.encode(len(response_bytes)))
+                        await stream.write(response_bytes)
+                        logger.debug("Sent PUT_VALUE acknowledgement")
+                    else:
+                        logger.warning("Invalid PUT_VALUE message format")
+
```

Comment:
Again, it seems like there should be something raised or other message returned to the other node or user if it fails, not just logging. Please excuse my question if this is not necessary or handled somewhere else.